adapters/between/between.go

```diff
@@ -3,169 +3,223 @@ package between
-			Method:  "POST",
-			Uri:     url,
-			Body:    reqJson,
-			Headers: headers}
-
-		requests = append(requests, &request)
+	var betweenExt openrtb_ext.ExtImpBetween
+	if err := json.Unmarshal(bidderExt.Bidder, &betweenExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("ignoring imp id=%s, invalid ImpExt", imp.ID),
+		}
 	}
 
-	return requests, errs
+	return &betweenExt, nil
 }
 
-func (a *BetweenAdapter) MakeBids(
-	internalRequest *openrtb.BidRequest,
-	externalRequest *adapters.RequestData,
-	response *adapters.ResponseData,
-) (*adapters.BidderResponse, []error) {
-
-	if response.StatusCode == http.StatusNoContent {
-		return nil, nil
+func (a *BetweenAdapter) buildEndpointURL(e *openrtb_ext.ExtImpBetween) (string, error) {
+	missingRequiredParameterMessage := "required BetweenSSP parameter \"%s\" is missing"
+	if e.Host == "" {
+		return "", &errortypes.BadInput{
+			Message: fmt.Sprintf(missingRequiredParameterMessage, "host"),
+		}
+	}
+	if e.PublisherID == "" {
+		return "", &errortypes.BadInput{
+			Message: fmt.Sprintf(missingRequiredParameterMessage, "publisher_id"),
+		}
 	}
+	return macros.ResolveMacros(a.EndpointTemplate, macros.EndpointTemplateParams{Host: e.Host, PublisherID: e.PublisherID})
+}
 
-	if response.StatusCode == http.StatusBadRequest {
-		return nil, []error{&errortypes.BadInput{
-			Message: fmt.Sprintf("Unexpected status code: %d. Bad request to dsp", response.StatusCode),
-		}}
-	} else if response.StatusCode != http.StatusOK {
-		return nil, []error{&errortypes.BadServerResponse{
-			Message: fmt.Sprintf("ERR, response with status %d", response.StatusCode),
-		}}
+func buildImpBanner(imp *openrtb.Imp) error {
+	if imp.Banner == nil {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("Request needs to include a Banner object"),
+		}
 	}
 
-	var bidResp openrtb.BidResponse
+	bannerCopy := *imp.Banner
+	banner := &bannerCopy
 
-	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
-		return nil, []error{err}
+	if banner.W == nil && banner.H == nil {
+		if len(banner.Format) == 0 {
+			return &errortypes.BadInput{
+				Message: fmt.Sprintf("Need at least one size to build request"),
+			}
+		}
+		format := banner.Format[0]
+		banner.Format = banner.Format[1:]
+		banner.W = &format.W
+		banner.H = &format.H
+		imp.Banner = banner
 	}
 
-	adapterResponse := adapters.NewBidderResponse()
-	adapterResponse.Currency = bidResp.Cur
+	return nil
+}
 
-	for _, seatBid := range bidResp.SeatBid {
-		for i := 0; i < len(seatBid.Bid); i++ {
-			bid := seatBid.Bid[i]
-			adapterResponse.Bids = append(adapterResponse.Bids, &adapters.TypedBid{
-				Bid:     &bid,
-				BidType: getMediaType(bid.ImpID, internalRequest.Imp),
-			})
-		}
+// Add Between required properties to Imp object
+func addImpProps(imp *openrtb.Imp, secure *int8, betweenExt *openrtb_ext.ExtImpBetween) {
+	imp.Secure = secure
+	bidFloor, err := strconv.ParseFloat(betweenExt.BidFloor, 64)
+	if err != nil {
```

Comment:
Please consider simplifying this as follows:
```
if bidFloor, err := strconv.ParseFloat(betweenExt.BidFloor, 64); err != nil || bidFloor <= 0 {
	imp.BidFloor = defaultBidfloor
} else {
	imp.BidFloor = bidFloor
}
```

Reply:
fixed